runelite-mixins/src/main/java/net/runelite/mixins/RSGrandExchangeOfferMixin.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2018, SomeoneWithAnInternetConnection
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.mixins;
+
+import net.runelite.api.GrandExchangeOfferState;
+import static net.runelite.api.GrandExchangeOfferState.BOUGHT;
+import static net.runelite.api.GrandExchangeOfferState.BUYING;
+import static net.runelite.api.GrandExchangeOfferState.CANCELLED;
+import static net.runelite.api.GrandExchangeOfferState.EMPTY;
+import static net.runelite.api.GrandExchangeOfferState.SELLING;
+import static net.runelite.api.GrandExchangeOfferState.SOLD;
+import net.runelite.api.mixins.Inject;
+import net.runelite.api.mixins.Mixin;
+import net.runelite.rs.api.RSGrandExchangeOffer;
+
+@Mixin(RSGrandExchangeOffer.class)
+public abstract class RSGrandExchangeOfferMixin implements RSGrandExchangeOffer
+{
+
+
+	@Inject
+	@Override
+	public GrandExchangeOfferState getState()
+	{
+		/*
+		 Internally a GrandExchangeOffer's state is represented as 4 flags
+		 packed into the lower half of a byte. They are:
+		*/
+
+		//Set for sell offers, unset for buy offers
+		int IS_SELLING = 1 << 3; // 0b1000
```

Comment:
These can be private static final. Other mixins rely on the compiler inlining private static final ints, which does seem to work reliably (although I think not guaranteed)